src/types.jl

```diff
@@ -535,10 +554,12 @@ end
 
 unflatten(t) = t
 
-function TermInterface.maketerm(::Type{<:BasicSymbolic}, head, args, type, metadata)
-    basicsymbolic(head, args, type, metadata)
+function TermInterface.maketerm(T::Type{<:BasicSymbolic}, head, args, metadata)
+    basicsymbolic(head, args, symtype(T), metadata)
 end
 
+symtype(::Type{<:Symbolic{T}}) where T = T
```

Comment:
@bowenszhu would like to hear your opinion on this one. Probably not best to have it called `symtype`, but `_symtype_t`?

Reply:
`symtype` is OK.